src/debug/telemetry/telemetry2_slot.h

```diff
@@ -0,0 +1,72 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#ifndef _ZEPHYR_SOC_INTEL_ADSP_TELEMETRY2_SLOT
+#define _ZEPHYR_SOC_INTEL_ADSP_TELEMETRY2_SLOT
+
+/*
+ * Telemetry2 slot is taken from SRAM window divided into several
+ * chunks with simple header declared bellow.
+ *
+ * "type" describes the contents of the chunk
+ * "size" describes the size of the chunk in bytes
+ *
+ * After the chunk another chunk is expected to follow, if there is
+ * enough space for another header. If type is 0 and size 0 it
+ * indicated that the rest of the slot is unused, and this header is
+ * free for taking.
+ *
+ * Each chunk is aligned to start at 64-byte cache line boundary.
+ *
+ * For example:
+ *      --------------------------------------------------  ---
+ *      | magic = TELEMETRY2_PAYLOAD_MAGIC               |   |
+ *      | hdr_size = 64                                  |   |
+ *      | total_size = 320                               |   |
+ *      | abi = TELEMETRY2_PAYLOAD_V0_0                  |  64 bytes
+ *      | tstamp = <aligned with host epoch>             |   |
+ *      | <padding>                                      |   |
+ *	--------------------------------------------------  ---
+ *	| type = ADSP_DW_TELEMETRY2_ID_THREAD_INFO       |   |
+ *      | size = 256                                     |  256 bytes
+ *	|    chunk data                                  |   |
+ *	|    ...                                         |   |
+ *	--------------------------------------------------  ---
+ *	| type = ADSP_DW_TELEMETRY2_TYPE_EMPTY           |
+ *      | size = 0                                       |
+ *	--------------------------------------------------
+ *
+ * The above depicts a telemetry2 slot with thread analyzer data
+ * of 256 bytes (including the header) in the first chunk and
+ * the rest of the slot being free.
+ */
+
+#include <stdint.h>
+
+#define TELEMETRY2_PAYLOAD_MAGIC 0x1ED15EED
+
+#define TELEMETRY2_PAYLOAD_V0_0 0
+
+/* Telemetry2 pauload hdr */
+struct telemetry2_payload_hdr {
+	uint32_t magic;		// used to identify valid data
+	uint32_t hdr_size;	// size of this header only in bytes
+	uint32_t total_size;	// total size of the whole payload in bytes
```

Comment:
including the header?

Reply:
The payload header is only there in the beginning of the telemetry2 slot. I add some comments above. This is a recent addition from Liam's suggestion. I just changed it a bit so that the payload header can be treated as just another chunk.